src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,80 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.RegistrationFailed;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int CORRECT_AGE = 19;
+    private static final String CORRECT_LOGIN = "testlogin";
+    private static final String CORRECT_PASSWORD = "testpassword";
+    private static final String SAME_TEST_LOGIN = "testlogin123";
+    private RegistrationServiceImpl registrationService;
```

Comment:
According to the checklist, consider whether it's better to use `@BeforeEach` or `@BeforeAll` for initializing your `RegistrationServiceImpl` instance. The choice between the two should be based on whether the instance needs to be recreated before each test (`@BeforeEach`) or can be reused (`@BeforeAll`).